### PR TITLE
shelter: add modal and filter icon (Dev-1080)

### PR DESCRIPTION
**Pull request by tglaz**
### Filter Shelter Results
https://betterangels.atlassian.net/browse/DEV-1080

* click on filter Icon -> see modal
* basic implementation

## Summary by Sourcery

Implement a filter modal in the shelter search component, allowing users to filter shelter results. Enhance the AddressAutocomplete component with a dynamic placeholder feature. Update the Tailwind CSS configuration to include new animations for modal transitions.

New Features:
- Introduce a filter modal that appears when the filter icon is clicked in the shelter search component.

Enhancements:
- Add a placeholder prop to the AddressAutocomplete component to allow dynamic placeholder text.

Build:
- Add new keyframes and animation for slide-in effects in the Tailwind CSS configuration.

### CODE REVIEW

File: apps/shelter-web/src/app/shared/components/address/AddressAutocomplete.tsx [hunk 1]
```diff
@@ -14,12 +14,18 @@ const boundsLA = getPlacesBounds({
...
-  const { onPlaceSelect, countryRestrictions = 'us', className = '' } = props;
+  const {
+    onPlaceSelect,
+    countryRestrictions = 'us',
+    placeholder,
+    className = '',
+  } = props;
 
   const [inputValue, setInputValue] = useState<string>('');
   const [sessionToken, setSessionToken] = useState<
```
[the remark this answers is not in the document]
Comment (tglaz) [reply apps/shelter-web/src/app/shared/components/address/AddressAutocomplete.tsx:31]: great. and thx.
but not related to this PR

File: libs/assets/src/icons/svg/filterOutline.svg [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    xmlns="http://www.w3.org/2000/svg">
+    <g id="cuida:filter-outline">
+        <g id="Group">
+            <path id="Vector" fill-rule="evenodd" clip-rule="evenodd" d="M3 7C3 6.73478 3.10536 6.48043 3.29289 6.29289C3.48043 6.10536 3.73478 6 4 6H20C20.2652 6 20.5196 6.10536 20.7071 6.29289C20.8946 6.48043 21 6.73478 21 7C21 7.26522 20.8946 7.51957 20.7071 7.70711C20.5196 7.89464 20.2652 8 20 8H4C3.73478 8 3.48043 7.89464 3.29289 7.70711C3.10536 7.51957 3 7.26522 3 7ZM5 11.5C5 11.2348 5.10536 10.9804 5.29289 10.7929C5.48043 10.6054 5.73478 10.5 6 10.5H18C18.2652 10.5 18.5196 10.6054 18.7071 10.7929C18.8946 10.9804 19 11.2348 19 11.5C19 11.7652 18.8946 12.0196 18.7071 12.2071C18.5196 12.3946 18.2652 12.5 18 12.5H6C5.73478 12.5 5.48043 12.3946 5.29289 12.2071C5.10536 12.0196 5 11.7652 5 11.5ZM8 16C8 15.7348 8.10536 15.4804 8.29289 15.2929C8.48043 15.1054 8.73478 15 9 15H15C15.2652 15 15.5196 15.1054 15.7071 15.2929C15.8946 15.4804 16 15.7348 16 16C16 16.2652 15.8946 16.5196 15.7071 16.7071C15.5196 16.8946 15.2652 17 15 17H9C8.73478 17 8.48043 16.8946 8.29289 16.7071C8.10536 16.5196 8 16.2652 8 16Z" fill="currentColor"/>
+        </g>
```
Comment (tglaz) [inline libs/assets/src/icons/svg/filterOutline.svg:6]: @Davit-BetterAngels @vecchp @mikefeldberg 

FYI - whenever adding an `svg` file to `libs/assets` we'd typically want the ability to change the color with CSS. To do so we'd need to make sure that `fill="currentColor"`. 

Typically the SVGs from Figma will have a hard-coded color, which we typically don't want (except for branded stuff etc)

File: libs/assets/src/icons/svg/filterOutline.svg
[the remark this answers is not in the document]
Comment (tglaz) [reply libs/assets/src/icons/svg/filterOutline.svg:1]: but yeah, Yuko has been working on reviewing the icons used here, so we will need to revisit all of these once she's done to have an approve lib of icons, and any new icon will require a file + licensing info. 
cause personally I don't ever wanna check the licensing after that

File: apps/shelter-web/src/app/shared/components/shelters/shelterSearch.tsx [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
 import { locationAtom } from '../../atoms/locationAtom';
+import { modalContentAtom } from '../../atoms/modalContentAtom';
+import { ModalAnimationEnum } from '../../modal/modal';
 import { AddressAutocomplete } from '../address/AddressAutocomplete';
 import { toGoogleLatLng } from '../map/utils/toGoogleLatLng';
 import { SheltersDisplay } from './sheltersDisplay';
 
 export function ShelterSearch() {
   const [location, setLocation] = useAtom(locationAtom);
+  const [_modal, setModal] = useAtom(modalContentAtom);
```
[the remark this answers is not in the document]
Comment (tglaz) [reply apps/shelter-web/src/app/shared/components/shelters/shelterSearch.tsx:14]: no real reason, and i'm good either way.

these var names are meant to be very local, so to me (at least) they need to be  clear in the local context.
but I agree that this thinking may also lead to things like  `const [a, setA] = ...` type of a thing.
but happy to  update

[the remark this answers is not in the document]
Comment (tglaz) [reply apps/shelter-web/src/app/shared/components/shelters/shelterSearch.tsx]: ended up renaming `modalContentAtom` to `modalAtom`
`content` is actually a prop of the Atom, so may just be simpler this way 🤷

File: apps/shelter-web/src/app/shared/modal/modalMask.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    'fixed',
+    'z-max',
+    'flex',
+    'justify-center',
+    'items-center',
+    'bg-gray-500/20 backdrop-blur',
+    className,
+  ].join(' ');
+
+  function onMaskClick() {
```
[the remark this answers is not in the document]
Comment (mikefeldberg) [reply apps/shelter-web/src/app/shared/modal/modalMask.tsx]: I actually had the same concern but wasn't sure if pressables (e.g., a checkbox) would also trigger a close

**Review by mikefeldberg — APPROVED** [review]

File: libs/assets/src/icons/svg/filterOutline.svg (hunk 2 above)
Comment (vecchp) [reply libs/assets/src/icons/svg/filterOutline.svg:6]: Yah, we've gone through that exercise a few times we either export it to not specify a color or manually edit the svg to have currentColor.

**Pull request merged**